src/CollectionClientDecorator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of JoliCode's Slack PHP API project.
+ *
+ * (c) JoliCode <coucou@jolicode.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JoliCode\Slack;
+
+use JoliCode\Slack\Api\Client;
+
+class CollectionClientDecorator
+{
+    const CURSOR_PAGINATION = [
+        'channelsList' => 'channels',
+        'conversationsHistory' => 'messages',
+        'conversationsList' => 'channels',
+        'conversationsMembers' => 'members',
+        'conversationsReplies' => 'messages',
+        'filesInfo' => 'comments',
+        'groupsList' => 'groups',
+        'imList' => 'ims',
+        'mpimList' => 'groups',
+        'reactionsList' => 'items',
+        'starsList' => 'items',
+        'usersList' => 'members',
+    ];
+
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    public function __call(string $name, array $arguments)
+    {
+        if (isset(self::CURSOR_PAGINATION[$name])) {
+            return $this->iterateOverCursor($name, $arguments);
+        }
+
+        throw new \BadMethodCallException('This client decorator should only be used with a collection endpoint');
```

Comment:
Maybe just run the query on the base client then?

Reply:
I started by doing that, but that did not work because I used a Generator typehint. But I think this should now work, so yes, I will change that